Makefile

```diff
@@ -36,9 +37,16 @@ endif
 
 LDFLAGS = -s -w -X ${PKG}/version.Version=${VERSION} -X ${PKG}/version.Revision=$(REVISION) -X ${PKG}/version.BuildTimestamp=$(BUILD_TIMESTAMP)
 
+OPTIMIZER_SERVER = tools/optimizer-server
+OPTIMIZER_SERVER_TOML = ${OPTIMIZER_SERVER}/Cargo.toml
+OPTIMIZER_SERVER_BIN = ${OPTIMIZER_SERVER}/target/release/optimizer-server
+
 .PHONY: build
 build:
 	GOOS=${GOOS} GOARCH=${GOARCH} ${PROXY} go build -ldflags "$(LDFLAGS)" -v -o bin/containerd-nydus-grpc ./cmd/containerd-nydus-grpc
+	${CARGO} fmt --manifest-path ${OPTIMIZER_SERVER_TOML} -- --check
+	${CARGO} build --release --manifest-path ${OPTIMIZER_SERVER_TOML} && cp ${OPTIMIZER_SERVER_BIN} ./bin
```

Comment:
Could we have a new Makefile target for building the `optimizer` ? It will even better if we have its own Makefile

Reply:
By default, we should not build the optimizer as developers may not have rust environment

Reply:
Yes, this is reasonable. Fixed it by adding `build-optimizer`. Is it acceptable?